CMakeLists.txt

```diff
@@ -58,13 +58,14 @@ STATIC_BUILD_WITH_DYNAMIC_STDLIB." OFF)
 OPTION (STATIC_BUILD_WITH_DYNAMIC_STDLIB "Build a (mostly) statically linked release binary. \
 Glibc and C++ standard library is linked dynamically." OFF)
 
-if (STATIC_BUILD_WITH_DYNAMIC_GLIBC OR STATIC_BUILD_WITH_DYNAMIC_STDLIB)
+OPTION (ENABLE_LTO "Build with LTO (deprecated: use CMAKE_INTERPROCEDURAL_OPTIMIZATION)" OFF)
+
+if (STATIC_BUILD_WITH_DYNAMIC_GLIBC OR STATIC_BUILD_WITH_DYNAMIC_STDLIB OR ENABLE_LTO)
     # We want to optimize the binary size for a static release binary by default.
-    set (_ENABLE_LTO_DEFAULT ON)
+    set (CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
 else()
-    set (_ENABLE_LTO_DEFAULT OFF)
+    set (CMAKE_INTERPROCEDURAL_OPTIMIZATION OFF)
```

Comment:
Wouldn't this mess with the setting of `CMAKE_INTERPROCEDRAL_OPTIMIZATION` done on the commandline? Actually probably both branches would mess with it. The idea behind `_ENABLE_LTO_DEFAULT` was that it is still possible to do `STATIC_*` with LTO explicitly disabled, I'm worried this would no longer work.

Reply:
@vlstill command-line arguments take precedence over anything in CMakeLists.txt